### ISSUE TITLE
`selector-no-id` and SCSS variable interpolation

### ISSUE THREAD

**Issue by mrksbnch** [stylelint/stylelint#1035]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Stylelint mistakes SCSS variable interpolation (`#{}`) for an `#id` selector.

> Which rule, if any, is this issue related to?

`selector-no-id`

> What (S)CSS is needed to reproduce this issue?

```sass
$class-type: ".";
$class-name: "test";

#{$class-type + $class-name} {
    color: #fff;
}
```

_Note:_ It only happens if there is a `+` character inside the SCSS interpolation brackets (`#{}`). So the following SCSS does not flag a warning:

```sass
$class-name: ".test";

#{$class-name} {
    color: #fff;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-id": true
  }
}
```

> Which version of stylelint are you using?

5.3.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint PATH --syntax scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to the SCSS syntax (variable interpolation `#{$variable}`)

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.scss
4:1	Unexpected id selector (selector-no-id)
```

**Comment by davidtheclark** [comment]
Duplicate of #945
